client/gui/InterfaceObjectConfigurable.cpp

```diff
@@ -190,18 +190,29 @@ ColorRGBA InterfaceObjectConfigurable::readColor(const JsonNode & config) const
 	logGlobal->debug("Reading color");
 	if(!config.isNull())
 	{
-		if(config.String() == "yellow")
-			return Colors::YELLOW;
-		if(config.String() == "white")
-			return Colors::WHITE;
-		if(config.String() == "gold")
-			return Colors::METALLIC_GOLD;
-		if(config.String() == "green")
-			return Colors::GREEN;
-		if(config.String() == "orange")
-			return Colors::ORANGE;
-		if(config.String() == "bright-yellow")
-			return Colors::BRIGHT_YELLOW;
+		if(config.isString())
+		{
+			if(config.String() == "yellow")
+				return Colors::YELLOW;
+			if(config.String() == "white")
```

Comment:
One of the things I hate about C++: a simple thing like using string values in enum, has no support in language :(

Could this be done separately? Where the enum is defined?

A map of some sorts 
Or a method to enum class...whith the switches inside
FromString(const std:string &enumValue)

Reply:
This is not enum, but a set of static constants. I guess in theory we also add fromString method in there, but so far this is the only place where we use string->color conversion so probably no need.